packages/react-instantsearch-dom/src/components/SearchBox.js

```diff
@@ -82,8 +82,7 @@ class SearchBox extends Component {
     isSearchStalled: PropTypes.bool,
     showLoadingIndicator: PropTypes.bool,
 
-    // For testing purposes
-    __inputRef: PropTypes.func,
+    inputRef: PropTypes.func,
```

Comment:
Since we expose this prop now, users can use new React APIs like [`createRef`](https://reactjs.org/docs/react-api.html#reactcreateref) or [`useRef`](https://reactjs.org/docs/hooks-reference.html#useref) that do not return functions—they will therefore get a PropType error. This should be more correct now:

```suggestion
    inputRef: PropTypes.oneOfType([
      PropTypes.func,
      PropTypes.shape({ current: PropTypes.instanceOf(HTMLInputElement) }),
    ]),
```

Reply:
The ref's already created inside this `SearchBox` here:

https://github.com/algolia/react-instantsearch/blob/26b8c682dde4587bdbf70caae6121f22e95b49cb/packages/react-instantsearch-dom/src/components/SearchBox.js#L252-L253

via 

https://github.com/algolia/react-instantsearch/blob/26b8c682dde4587bdbf70caae6121f22e95b49cb/packages/react-instantsearch-dom/src/components/SearchBox.js#L133-L138

So users should just provide a callback that will get passed the ref.  

If it would be better to ask users to create and pass in refs, I can do that instead.

Reply:
I see, thanks for the context. Yeah, we should support native React refs so that the API is as intuitive as using a basic React component.

Reply:
@francoischalifour The latest commit should cover both cases.

One thing I was not sure about: `HTMLInputElement`, is that going to be okay for server side rendering, even though it's a browser global?  I tried simply copying this PR's version `SearchBox` into a Gatsby project I'm working on and ran into `WebpackError: ReferenceError: HTMLInputElement is not defined`.  Is that dealt with somehow?

Reply:
The CI was complaining about `HTMLInputElement`, so l investigated.  Turns out `React.createRef` just makes: `{ current: null }`, so I changed the PropType to look for, basically, that.

Reply:
That looks good—thanks for the investigation!